### PR TITLE
Remove unused code

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

**Review by dcousens — APPROVED** [review]
utACK

**Comment by NicolasDorier** [comment]
utACK

File: src/dbwrapper.h [hunk 1]
```diff
@@ -132,10 +132,6 @@ class CDBIterator
         return true;
     }
 
-    unsigned int GetKeySize() {
```
Comment (paveljanik) [inline src/dbwrapper.h]: `GetKeySize` is referenced in the comment in `src/coins.h`.

Comment (practicalswift) [inline src/dbwrapper.h]: @paveljanik Good point! Reference in `src/coins.h` removed. Squashed and pushed as well :-)

**Comment by JeremyRubin** [comment]
I'd suggest reverting interpreter.cpp, unless anyone strongly objects. Seeing those (albeit unused) static consts helped me to understand the script code more quickly.

**Comment by practicalswift** [comment]
@JeremyRubin Perhaps I should add them back but commented out. Sounds good? :-)

**Comment by laanwj** [comment]
> I'd suggest reverting interpreter.cpp, unless anyone strongly objects. Seeing those (albeit unused) static consts helped me to understand the script code more quickly.

Agree, let's just put those back. There is zero overhead to defining a few extra constants for clarity even if unused.

File: src/wallet/db.h [hunk 2]
```diff
@@ -300,12 +300,6 @@ class CDB
         return (ret == 0);
     }
 
-    bool ReadVersion(int& nVersion)
```
Comment (laanwj) [inline src/wallet/db.h]: We probably want to keep this one around for symmetry.

**Comment by jonasschnelli** [comment]
utACK 20da129626849511b41a870333dd0144eeaf0c6d, agree with @laanwj about `ReadVersion()`.

File: src/chainparams.h [hunk 3]
```diff
@@ -61,8 +61,6 @@ class CChainParams
     int GetDefaultPort() const { return nDefaultPort; }
 
     const CBlock& GenesisBlock() const { return genesis; }
-    /** Make miner wait to have peers to avoid wasting work */
-    bool MiningRequiresPeers() const { return fMiningRequiresPeers; }
```
Comment (MarcoFalke) [inline src/chainparams.h]: Either keep everything or remove the member as well, no?

**Review by maflcko — COMMENTED** [review]
ACK on removing the clearly outdated code (such as in test/) Not sure about the gains on removing some of the other methods.

**Comment by practicalswift** [comment]
@JeremyRubin @laanwj The changes to `interpreter.cpp` have now been reverted as suggested.

@laanwj @jonasschnelli `ReadVersion(int& nVersion)` is now kept around for symmetry.

@MarcoFalce Thanks for noticing! `fMiningRequiresPeers` has now been removed as well :-)

Thanks for reviewing! Please let me know if any further adjustments are needed.

**Review by dcousens — APPROVED** [review]
re-ACK

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/9987/commits/8dc957ae06366d023e0d23c92279c5fb04b515d0

**Pull request merged**